src/runtime/threads/detail/thread_pool.cpp

```diff
@@ -43,6 +44,14 @@
 
 namespace hpx { namespace threads { namespace detail
 {
+    thread_id_type& background_thread()
+    {
+        struct tls_tag {};
+        static hpx::util::thread_specific_ptr<thread_id_type, tls_tag> thrd;
```

Comment:
Why is the background thread stored in a thread-local variable. Wouldn't be sufficient to have it on the stack as before?